src/main/java/core/basesyntax/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction, Map<String, Integer> inventory) {
+        int currentQuantity = inventory.getOrDefault(transaction.getFruit(), 0);
+        if (currentQuantity < transaction.getQuantity()) {
```

Comment:
Before performing the purchase operation, you should check if the fruit is available in the inventory. If the `getOrDefault` method returns 0, it implies that the fruit is not present, and a purchase operation should not be allowed.